dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityInstanceStore.java

```diff
@@ -826,14 +821,14 @@ else if ( params.isEventStatus( EventStatus.SKIPPED ) )
     @Override
     public boolean exists( String uid )
     {
-        Integer result = jdbcTemplate.queryForObject( "select count(*) from trackedentityinstance where uid=? and deleted is false", Integer.class, uid );
+        Integer result = jdbcTemplate.queryForObject( "select count(0) from trackedentityinstance where uid=? and deleted is false", Integer.class, uid );
         return result != null && result > 0;
     }
 
     @Override
     public boolean existsIncludingDeleted( String uid )
     {
-        Integer result = jdbcTemplate.queryForObject( "select count(*) from trackedentityinstance where uid=?", Integer.class, uid );
+        Integer result = jdbcTemplate.queryForObject( "select count(0) from trackedentityinstance where uid=?", Integer.class, uid );

```

Comment:
Good that you asked. I was going to explain to you the difference and provide a link. But then when I read it, it is exactly the opposite as I remembered it. I thought `count(0)` is faster than `count(*)`. But it is the opposite. Here is the article: https://www.citusdata.com/blog/2016/10/12/count-performance/  Only the top part and table on the bottom is relevant for this topic.

Will fix it back to `count(0)`

Reply:
Please, be aware that this method uses a `JdbcTemplate` in a class that is supposed to use Hibernate only. This may lead to data corruption, because the result of the jdbc query may not be aware of data that are still in the Hibernate session and are not flushed yet.
I recommend you use `getSession().createNativeQuery("...")`

Reply:
Thanks Luciano. Fixed.